src/watchers/html.ts

```diff
@@ -251,6 +251,51 @@ export class HTMLWatcher extends DOMWatcher<Instance> {
     };
   }
 
+  _mouseMoveEventHandler(tagname:string) {
+    return (event:MouseEvent) => {
+      let {target} = event;
+      if (!this.isInstance(target)) return;
+
+      let currentTarget:Element|null = target;
+
+      while (currentTarget && this.isInstance(currentTarget)) {
+        let eavs:(RawEAV|RawEAVC)[] = [];
+        let elemId = currentTarget.__element!;
+
+        if (currentTarget.listeners && currentTarget.listeners["mouse-position"]) {
+          let eventId = createId();
+          eavs.push(
+            [eventId, "tag", "html/event"],
+            [eventId, "tag", `html/event/${tagname}`],
+            [eventId, "page-x", event.pageX],
+            [eventId, "page-y", event.pageY],
+            [eventId, "window-x", event.clientX],
+            [eventId, "window-y", event.clientY],
+
+            [eventId, "target", elemId]
+          );
+
+          let current:Element|null = currentTarget;
```

Comment:
We should instead do

```
currentTarget = currentTarget.parentNode
```

This way we only have a single loop, and visit each element in the hierarchy once. This will also give us a separate event per element, which gives us the flexibility to add `local-x` and `local-y` attributes to the mose-position event.